### ISSUE TITLE
Authentication workflow problem when email confirmation is required.

### ISSUE THREAD

**Issue by Eilon** [aspnet/Identity#1667]
_From @Eilon on October 24, 2017 21:48_

_From @Bartmax on October 21, 2015 8:23_

**TL;DR:**
User using external login provider and application with email confirmation required will be able to login the first time (on user creation) without confirming email and he/she won't be able to log in again.

---

When a visitor enter the page and hit login, then choosing an external provider, it's then redirected to username/email confirmation.

typing an email creates and account and log the user in. (user should be required to confirm email)
There's no email sending code in the ExternalLogin workflow.

if user logout and then try to login again, it's again presenting him the Associate account.

If he types in the same email address it says it's already taken.
If he types a new email address, it says Invalid login (already associated).

If he chooses to forgot password, then it's presented with a forgot password confirmation but not email sent because it's not confirmed.

Problems summary:
1. There's no way a user can log in (for a second time) in this scenario.
2. The user can use the system without email confirmation as long as he/she stays logged in when account was created.
3. No email confirmation is sent on this workflow.
4. Forgot password does **not** help.

_Copied from original issue: aspnet/Templates#242_

_Copied from original issue: aspnet/templating#108_

**Comment by Eilon** [comment]
_From @rustd on February 18, 2016 19:2_

The user should not get an Associate form when they login.

**Comment by Eilon** [comment]
FYI this bug is from 2015... so take that into account.

**Comment by javiercn** [comment]
This is still an issue. I filed a similar issue not too long ago. I'll point this one to that one and close it.
https://github.com/aspnet/Identity/issues/1637
